Beeper/BarcelonaMautrixIPC/Runtime/MautrixIPCChannel.swift

```diff
@@ -0,0 +1,136 @@
+//
+//  MautrixImessageIPCChannel.swift
+//  BarcelonaMautrixIPC
+//
+//  Created by Brad Murray on 2022-11-29.
+//
+
+import Barcelona
+import Combine
+import ERBufferedStream
+import Foundation
+
+let TERMINATOR = Data("\n".utf8)
+
+public class MautrixIPCChannel {
+    public var receivedPayloads = PassthroughSubject<IPCPayload, Never>()
+    
+    private let inputHandle: FileHandle
+    private let outputHandle: FileHandle
+    
+    let sharedBarcelonaStream: ERBufferedStream<IPCPayload> = {
+        let stream = ERBufferedStream<IPCPayload>()
+        stream.decoder.dateDecodingStrategy = .iso8601
+        return stream
+    }()
+    
+    var pongedOnce = false
+    
+    private var cancellables = Set<AnyCancellable>()
+    
+    public init(inputHandle: FileHandle, outputHandle: FileHandle) {
+        self.inputHandle = inputHandle
+        self.outputHandle = outputHandle
+        
+        // Set up our reading pipeline
+        
+        inputHandle.handleDataAsynchronously(sharedBarcelonaStream.receive(data:))
+        
+        // unsafeBitCast is needed because sharedBarcelonaStream is actually an OpenCombine stream as opposed to a
+        // Combine stream, but it still has a store property we can call
+        unsafeBitCast(sharedBarcelonaStream.subject
```

Comment:
Can we not just make `cancellables` store a set of whatever `OpenCombineStream.sink` returns? Do we really need to cast it?